src/controller/CHIPDeviceController.cpp

```diff
@@ -90,6 +93,10 @@ using namespace chip::Encoding;
 #if CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONER_DISCOVERY
 using namespace chip::Protocols::UserDirectedCommissioning;
 #endif // CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONER_DISCOVERY
+#if CHIP_DEVICE_CONFIG_ENABLE_WIFIPAF
+static std::shared_ptr<DeviceCommissioner> DevCommPtr;
```

Comment:
I don't understand how this is supposed to work.  Per C++ spec:
```
If the object pointed to by ptr is already owned, the function generally results in undefined behavior. 
```
and that's the scenario here, no?  There are non-shared-ptr bit that will happily delete that object out from under you.